File: src/main/java/org/cubexmc/metro/model/DirectionMode.java
package org.cubexmc.metro.model;

public enum DirectionMode {
    BI_DIRECTIONAL,
    CIRCULAR,
    SINGLE_DIRECTION;

    public static DirectionMode from(String s, DirectionMode def) {
        if (s == null)
            return def;
        try {
            return valueOf(s.toUpperCase());
        } catch (IllegalArgumentException e) {
            return def;
        }
    }
}

File: src/main/java/org/cubexmc/metro/model/DirectionMode.kt
package org.cubexmc.metro.model

import java.util.Locale

enum class DirectionMode {
    BI_DIRECTIONAL,
    CIRCULAR,
    SINGLE_DIRECTION,
    ;

    companion object {
        @JvmStatic
        fun from(s: String?, def: DirectionMode): DirectionMode {
            if (s == null) {
                return def
            }
            return try {
                valueOf(s.uppercase(Locale.getDefault()))
            } catch (_: IllegalArgumentException) {
                def
            }
        }
    }
}
